### DISCUSSION TITLE
M2M authentication with auth0-client

### DISCUSSION THREAD

**Discussion by fabioS24** [mia-platform/community#498] [Q&A] [answered]
### What product are you having troubles with?

Marketplace

### What Console version are you using?

v12.4.0

### Description

Hi everyone,  
  
Does auth0-client support machine to machine authentication?  

In case I want to perform M2M authentication together with the username-password authentication provided by the auth0-client, are there any guidelines to follow?

I saw that the client-credentials service allows you to authenticate with client credentials grant but I fear that it may conflict with the auth0-client, as both expose the same endpoint for authentication. Is there a recommended way for the two services to coexist?

Would it be possible to expand the auth0-client to implement M2M authentication using the Auth0 API? So as to be able to use both types of authentication (username-password and M2M) with a single service.

Thank you

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, as  of now the auth0-client does not support m2m authentication
